dbms/src/Storages/StorageDeltaMerge.h

```diff
@@ -14,29 +14,24 @@
 
 #pragma once
 
+#include <Common/Logger.h>
```

Comment:
Can we remove this header file？

Reply:
It should be a better practice to include everything you used in this header file, even if it may be included transitively by other included header files. In this way, this header is less likely to break when other headers are refactored.

Reply:
We can use forward declaration here, using
```
namespace DB
{
class Logger;
using LoggerPtr = std::shared_ptr<Logger>;
}
```
instead of including the `<Common/Logger.h>`.

Here I think both are OK since `Common/Logger.h` itself is a clean header file that doesn't bring many other header files.